src/error_chain.rs

```diff
@@ -137,6 +137,17 @@ macro_rules! error_chain_processed {
                 )
             }
 
+            /// Construct a chained error from another boxed error and a kind, and generates a backtrace
+            pub fn with_boxed_chain<K>(error: Box<::std::error::Error + Send>, kind: K)
+                -> $error_name
+                where K: Into<$error_kind_name>
+            {
+                $error_name(
+                    kind.into(),
+                    $crate::State::new::<$error_name>(error, ),
+                )
+            }
+
```

Comment:
Hey Yamakaky,
I've been working with you on this PR for a long time. I'd really like to improve error_chain to make it more featureful, but I'm getting the impression that you're not planning to accept the `with_boxed_chain` method. Could you tell me where this is going? If this was 2 commits, would you merge it? I have production code depending on fixing these issues, and if you think we're going in different directions, I'd like to know that.

Thank you!